src/rowbinary/ser.rs

```diff
@@ -134,12 +134,20 @@ impl<'a, B: BufMut> Serializer for &'a mut RowBinarySerializer<B> {
     #[inline]
     fn serialize_newtype_variant<T: Serialize + ?Sized>(
         self,
-        name: &'static str,
-        _variant_index: u32,
-        variant: &'static str,
-        _value: &T,
+        _name: &'static str,
+        variant_index: u32,
+        _variant: &'static str,
+        value: &T,
     ) -> Result<()> {
-        panic!("newtype variant types are unsupported: `{name}::{variant}`");
```

Comment:
Note: this code implicitly allows using enums at the top level. Usually, it's avoided by using either the parameter stored in the serializer struct or dedicated `SerializeStruct`/etc associated types.

In general, it's okay (the user will get some error anyway), but instead of more descriptive panic, it ends with "not enough data."

Also, it produces an unclear message for forgotten `serde_repr` (`Enum8` and `Enum16`).

It's not a blocker for merge because it's an error anyway. But, probably, we should leave "TODO" at least in the code about it.